docs/technical/concurrency.md

```diff
@@ -46,12 +48,19 @@ branch was moved to two different locations, they will appear in `jj log` in
 both locations but with a "?" after the name, and `jj status` will also inform
 the user about the conflict.
 
-The most important piece in the lock-free design is the "operation log". That is
-what allows us to detect and merge concurrent operations.
+Note that, for now, such use of Jujutsu is not currently thoroughly tested,
+especially in the context of [co-located
+ repositores](../glossary.md#co-located-repos). While the contents of commits
+should be safe, concurrent modification of a repository from different computers
+might, in the worst case, lose some branch pointers. Note that, unlike in pure
```

Comment:
Can it lose branch pointers?

Reply:
I don't know for sure, but I'm guessing so, since Git's branches don't understand conflicted states. So, if two `git` commands move branches to different places on two different computers at the same time, and then Dropbox syncs this in some crazy fashion, and then `jj` will tries to import that mess, who knows what will happen. jj could also try to import it in the middle of those operations.

This whole paragraph is an educated guess.

Reply:
Ah, I see what you mean. Yes, that seems like it could happen. It will still be available in the operation log, but I agree that it it's hard to predict what the latest imported state can be.